qiskit_machine_learning/utils/loss_functions/loss_functions.py

```diff
@@ -20,37 +20,80 @@
 
 class Loss(ABC):
     """
-    Abstract base class for Loss.
+    Abstract base class for computing Loss.
+
     """
 
-    def __call__(self, predict, target):
+    def __call__(self, predict: np.ndarray, target: np.ndarray) -> float:
+        """
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Returns:
+            a float value of the loss function
+        """
         return self.evaluate(predict, target)
 
     @abstractmethod
-    def evaluate(self, predict, target):
-        """evaluate"""
+    def evaluate(self, predict: np.ndarray, target: np.ndarray):
+        """
+        An abstract method for evaluating the loss function
+
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match
+        """
         raise NotImplementedError
 
     @abstractmethod
-    def gradient(self, predict, target):
-        """gradient"""
+    def gradient(self, predict: np.ndarray, target: np.ndarray):
+        """
+        An abstract method for computing the gradient
+
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match.
+        """
         raise NotImplementedError
 
     @staticmethod
-    def _validate(predict, target):
+    def _validate(predict: np.ndarray, target: np.ndarray):
```

Comment:
```suggestion
    def _validate(predict: np.ndarray, target: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
```
The typehint should have the return type. The docstring has a Returns entry so thats good.

I must admit this method is kinda confusing. If it just needs to validate the shapes why does it need to return them since the caller already has them? Now I see it makes a numpy array of them, but then the typehint says it takes a numpy array and not something like a list where I could see it adjusting the types as needed as well as doing some validation, with the (possibly) adjusted values returned,

Reply:
updated